pymc3/tests/test_sampling.py

```diff
@@ -121,6 +121,37 @@ def test_iter_sample(self):
             for i, trace in enumerate(samps):
                 assert i == len(trace) - 1, "Trace does not have correct length."
 
+    def test_sample_does_not_modify_start_as_list_of_dicts(self):
```

Comment:
Does this test fail in master? It looks like no transforms would be added in this test model and therefore the dictionary wouldn't be changed anyway, but I could be wrong. I imagined it worked something like your test below but with a dictionary for each chain:

```python
start_dict = [{"X0_mu": 25}, {"X0_mu": 25}]
with pm.model() as m:
   X0_mu = pm.Lognormal("X0_mu", mu=np.log(0.25), sd=0.10)
   trace = pm.sample(
                step=pm.Metropolis(),
                tune=5,
                draws=10,
                chains=2,
                start=start_dict,
            )
  assert start_dict == ...
```

Reply:
Oh I see, you made one parameter be missing on purpose in each chain...

Reply:
> Does this test fail in master? It looks like no transforms would be added in this test model and therefore the dictionary wouldn't be changed anyway.

unfortunately, it updates the dictionary with `model.test_point` even though no transformed variable is there to be added.